### PR TITLE
Fix release to maven central

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- added sources for all modules
- refactored creating publication for gradle plugin

### CODE REVIEW

File: diktat-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -38,6 +39,8 @@ tasks.withType<KotlinCompile> {
 }
 
 gradlePlugin {
+    website = "https://diktat.saveourtool.com/"
+    vcsUrl = "https://github.com/saveourtool/diktat"
     plugins {
         create("diktatPlugin") {
             id = "com.saveourtool.diktat.diktat-gradle-plugin"
```
Comment (nulls) [inline diktat-gradle-plugin/build.gradle.kts:46]:
```suggestion
            id = "com.saveourtool.diktat"
```

File: gradle/plugins/src/main/kotlin/com/saveourtool/diktat/buildutils/VersioningConfiguration.kt [hunk 2]
```diff
@@ -23,7 +24,7 @@ fun Project.configureVersioning() {
     // should be provided in the gradle.properties
     configure<ReckonExtension> {
         setDefaultInferredScope(Scope.MINOR.name)
-        snapshots()
+        stages("rc", "final")
```
Comment (nulls) [inline gradle/plugins/src/main/kotlin/com/saveourtool/diktat/buildutils/VersioningConfiguration.kt:27]: it breaks snapshot releases

File: gradle/plugins/src/main/kotlin/com/saveourtool/diktat/buildutils/kotlin-jvm-configuration.gradle.kts [hunk 3]
```diff
@@ -26,6 +26,11 @@ kotlin {
     }
 }
 
+tasks.register<Jar>(SOURCES_JAR) {
+    archiveClassifier.set("sources")
+    from(kotlin.sourceSets.main.map { it.kotlin })
```
Comment (nulls) [inline gradle/plugins/src/main/kotlin/com/saveourtool/diktat/buildutils/kotlin-jvm-configuration.gradle.kts:31]: `kotlin` is available only after applying `kotlin("jvm")`

File: gradle/plugins/src/main/kotlin/com/saveourtool/diktat/buildutils/VersioningConfiguration.kt (hunk 2 above)
Comment (nulls) [reply gradle/plugins/src/main/kotlin/com/saveourtool/diktat/buildutils/VersioningConfiguration.kt]: fixed

File: diktat-gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (nulls) [reply diktat-gradle-plugin/build.gradle.kts]: applyed

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**